schema/loopring/fn_process_block.sql

```diff
@@ -0,0 +1,161 @@
+DROP FUNCTION loopring.fn_process_block;
+
+DROP TYPE loopring.transaction_struct;
+DROP TYPE loopring.deposit_struct;
+DROP TYPE loopring.withdraw_struct;
+DROP TYPE loopring.transfer_struct;
+DROP TYPE loopring.spot_trade_struct;
+DROP TYPE loopring.account_update_struct;
+DROP TYPE loopring.amm_update_struct;
+DROP TYPE loopring.signature_verification_struct;
+
+CREATE TYPE loopring.deposit_struct AS (
+    toAddress bytea,
+    toAccount integer,
+    token integer,
+    amount double precision
+);
+
+CREATE TYPE loopring.withdraw_struct AS (
+    fromAddress bytea,
+    fromAccount integer,
+    token integer,
+    amount double precision,
+    feeToken integer,
+    fee double precision
+);
+
+CREATE TYPE loopring.transfer_struct AS (
+    token integer,
+    amount double precision,
+    feeToken integer,
+    fee double precision,
+    fromAccount integer,
+    toAccount integer,
+    toAddress bytea,
+    fromAddress bytea
+);
+
+CREATE TYPE loopring.spot_trade_struct AS (
+    accountA integer,
+    accountB integer,
+    tokenA integer,
+    tokenB integer,
+    amountA double precision,
+    amountB double precision
+);
+
+CREATE TYPE loopring.account_update_struct AS (
+    ownerAddress bytea,
+    ownerAccount integer,
+    feeToken integer,
+    fee double precision,
+    publicKey bytea
+);
+
+CREATE TYPE loopring.amm_update_struct AS (
+    ownerAddress bytea,
+    ownerAccount integer,
+    token integer,
+    feeBips integer,
+    weight double precision
+);
+
+CREATE TYPE loopring.signature_verification_struct AS (
+    ownerAddress bytea,
+    ownerAccount integer
+);
+
+CREATE TYPE loopring.transaction_struct AS (
+    block_timestamp timestamptz,
+    blockIdx integer,
+    txIdx integer,
+    txType integer,
+    deposit loopring.deposit_struct,
+    withdraw loopring.withdraw_struct,
+    transfer loopring.transfer_struct,
+    spot_trade loopring.spot_trade_struct,
+    account_update loopring.account_update_struct,
+    amm_update loopring.amm_update_struct,
+    signature_verification loopring.signature_verification_struct
+);
+
+
+CREATE OR REPLACE FUNCTION loopring.fn_process_block(blockSize integer, _data bytea, block_timestamp timestamptz, blockIdx integer)
```

Comment:
Should we rename the "old" process_block function `loopring.fn_process_trade_block` to something that indicates a version?